pubsub/README.md

```diff
@@ -18,121 +18,148 @@ google-cloud-pubsub = <version>
 
 ```rust
  use google_cloud_pubsub::client::Client;
- use google_cloud_gax::cancel::CancellationToken;
- use google_cloud_googleapis::pubsub::v1::PubsubMessage;
- use google_cloud_pubsub::topic::TopicConfig;
- use google_cloud_pubsub::subscription::SubscriptionConfig;
- use google_cloud_gax::grpc::Status;
- use tokio::task::JoinHandle;
-
- #[tokio::main]
- async fn main() -> Result<(), Status> {
-
-     // Create pubsub client.
-     // The default project is determined by credentials.
-     // - If the GOOGLE_APPLICATION_CREDENTIALS is specified the project_id is from credentials.
-     // - If the server is running on GCP the project_id is from metadata server
-     // - If the PUBSUB_EMULATOR_HOST is specified the project_id is 'local-project'
-     let mut client = Client::default().await.unwrap();
-
-     // Create topic.
-     let topic = client.topic("test-topic");
-     if !topic.exists(None, None).await? {
-         topic.create(None, None, None).await?;
-     }
-
-     // Start publisher.
-     let publisher = topic.new_publisher(None);
-
-     // Publish message.
-     let tasks : Vec<JoinHandle<Result<String,Status>>> = (0..10).into_iter().map(|_i| {
-         let publisher = publisher.clone();
-         tokio::spawn(async move {
-             let mut msg = PubsubMessage::default();
-             msg.data = "abc".into();
-             // Set ordering_key if needed (https://cloud.google.com/pubsub/docs/ordering)
-             // msg.ordering_key = "order".into();
-
-             // Send a message. There are also `publish_bulk` and `publish_immediately` methods.
-             let mut awaiter = publisher.publish(msg).await;
-             
-             // The get method blocks until a server-generated ID or an error is returned for the published message.
-             awaiter.get(None).await
-         })
-     }).collect();
-
-     // Wait for all publish task finish
-     for task in tasks {
-         let message_id = task.await.unwrap()?;
-     }
-
-     // Wait for publishers in topic finish.
-     let mut publisher = publisher;
-     publisher.shutdown();
-
-     Ok(())
- }
+use google_cloud_gax::cancel::CancellationToken;
+use google_cloud_googleapis::pubsub::v1::PubsubMessage;
+use google_cloud_pubsub::topic::TopicConfig;
+use google_cloud_pubsub::subscription::SubscriptionConfig;
+use google_cloud_gax::grpc::Status;
+use tokio::task::JoinHandle;
+
+#[tokio::main]
+async fn main() -> Result<(), Status> {
+
+    // Create pubsub client.
+    // The default project is determined by credentials.
+    // - If the GOOGLE_APPLICATION_CREDENTIALS is specified the project_id is from credentials.
+    // - If the server is running on GCP the project_id is from metadata server
+    // - If the PUBSUB_EMULATOR_HOST is specified the project_id is 'local-project'
+    let mut client = Client::default().await.unwrap();
+
+    // Create topic.
+    let topic = client.topic("test-topic");
+    match topic.exists(None, None).await {
```

Comment:
`if !topic.exists(None, None).await?` uses early returning. 
`topic.exists` returns false when the `NotFound` occurs
https://github.com/yoshidan/google-cloud-rust/blob/main/pubsub/src/topic.rs#L110